worker/machiner/machiner.go

```diff
@@ -191,7 +191,11 @@ func (mr *Machiner) Handle(_ <-chan struct{}) error {
 			logger.Tracef("machine still has storage attached")
 			return nil
 		}
-		return errors.Annotatef(err, "%s failed to set machine to dead", mr.config.Tag)
+		err = errors.Annotatef(err, "%s failed to set machine to dead", mr.config.Tag)
```

Comment:
Be careful here.
1. err = errors.Annotatef(err, "failed to set status")
is masking the error we really care about. It just needs to be logged as it's not the main issue.
2. We need to check the behaviour - this CodeMachineHasAttachedStorage etc error I think may be part of the normal cleanup - separate workers will be doing the job of removing units. We need to check if it's really appropriate to set an error here and hence stop the destroy CLI, or if it is expected this issue will be transient and go away when the other workers run, in which case we don't want to set status to error.